includes/Stories_Renderer/Renderer.php

```diff
@@ -0,0 +1,512 @@
+<?php
+/**
+ * Stories Renderer Base class.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Stories_Renderer;
+
+use Google\Web_Stories\Interfaces\Renderer as RenderingInterface;
+use Google\Web_Stories\Model\Story;
+use Google\Web_Stories\Story_Query as Stories;
+use Google\Web_Stories\Story_Post_Type;
+use Google\Web_Stories\Traits\Assets;
+use Iterator;
+
+/**
+ * Renderer class.
+ *
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ * @SuppressWarnings(PHPMD.TooManyPublicMethods)
+ * @implements Iterator<int, \WP_Post>
+ */
+abstract class Renderer implements RenderingInterface, Iterator {
+
+	use Assets;
+
+	/**
+	 * Web Stories stylesheet handle.
+	 *
+	 * @var string
+	 */
+	const STYLE_HANDLE = 'web-stories-list-styles';
+
+	/**
+	 * Stories object
+	 *
+	 * @var Stories Stories object
+	 */
+	protected $stories;
+
+	/**
+	 * Story attributes
+	 *
+	 * @var array An array of story attributes.
+	 */
+	protected $attributes = [];
+
+	/**
+	 * Story posts.
+	 *
+	 * @var array An array of story posts.
+	 */
+	protected $story_posts = [];
+
+	/**
+	 * Pointer to iterate over stories.
+	 *
+	 * @var int
+	 */
+	private $position = 0;
+
+	/**
+	 * Height for displaying story.
+	 *
+	 * @var string
+	 */
+	protected $height = '430';
+
+	/**
+	 * Width for displaying story.
+	 *
+	 * @var string
+	 */
+	protected $width = '285';
+
+	/**
+	 * Whether content overlay is enabled for story.
+	 *
+	 * @var bool
+	 */
+	protected $content_overlay;
```

Comment:
Should these be arguments to pass to `render()`?

Seems more intuitive to do:

```php
$renderer = new Renderer( $my_stories );
$renderer->render( [ 'width' => 9999, 'height' => 1234 ] );
```

Thoughts?

Reply:
I see you have changed this now... But why are width & height class properties in the first place?